streams/src/main/java/org/apache/kafka/streams/state/internals/SessionKeySchema.java

```diff
@@ -50,15 +50,17 @@ public HasNextCondition hasNextCondition(final Bytes binaryKey, final long from,
         return new HasNextCondition() {
             @Override
             public boolean hasNext(final KeyValueIterator<Bytes, ?> iterator) {
-                if (iterator.hasNext()) {
+                while (iterator.hasNext()) {
                     final Bytes bytes = iterator.peekNextKey();
                     final Bytes keyBytes = Bytes.wrap(SessionKeySerde.extractKeyBytes(bytes.get()));
-                    if (!keyBytes.equals(binaryKey)) {
-                        return false;
-                    }
                     final long start = SessionKeySerde.extractStart(bytes.get());
                     final long end = SessionKeySerde.extractEnd(bytes.get());
-                    return end >= from && start <= to;
+                    if (keyBytes.equals(binaryKey)
```

Comment:
minor performance nit, it might be worthwhile to have something like SessionKeySerde.inWindow(bytes, binaryKey, from, to) to avoid creating three bytebuffer objects for each comparison.